icu4c/source/i18n/usearch.cpp

```diff
@@ -222,6 +222,9 @@ inline int32_t * addTouint32_tArray(int32_t    *destination,
         int32_t *temp = (int32_t *)allocateMemory(
                                          sizeof(int32_t) * newlength, status);
         if (U_FAILURE(*status)) {
+            if (temp != NULL) {
```

Comment:
Same here. could we use
```
LocalMemory<int32_t> temp;
temp.allocateInsteadAndCopy(newlength);
```
to fix it?

Reply:
same as above

smart pointers good, but you could also unconditionally `uprv_free(temp)`

Reply:
@FrankYFTang @markusicu like this?:
```diff
--- a/icu4c/source/i18n/usearch.cpp
+++ b/icu4c/source/i18n/usearch.cpp
@@ -199,6 +199,7 @@ inline int32_t * addTouint32_tArray(int32_t    *destination,
         int32_t* temp = static_cast<int32_t*>(allocateMemory(
                                          sizeof(int32_t) * newlength, status));
         if (U_FAILURE(*status)) {
+            uprv_free(temp);
             return nullptr;
         }
         uprv_memcpy(temp, destination, sizeof(int32_t) * (size_t)offset);
@@ -240,6 +241,7 @@ inline int64_t * addTouint64_tArray(int64_t    *destination,
                                          sizeof(int64_t) * newlength, status));
 
         if (U_FAILURE(*status)) {
+            uprv_free(temp);
             return nullptr;
         }
 
```